Jetpack Cloud: Add loading screen logo and animation

delawski opened the pull request:
This PR adds a pulsating Jetpack logo when the Jetpack Cloud is being loaded:

![2020-01-31 12 36 56](https://user-images.githubusercontent.com/478735/73536654-aa9bba80-4426-11ea-806a-63ba4895c0a4.gif)

#### Changes proposed in this Pull Request

* Add animated Jetpack logo to the Jetpack Cloud loading screen 

#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Start (or re-run) Calypso build process with: `CALYPSO_ENV=jetpack-cloud-development npm start`
* In devtools set network throttling to Slow 3G so that it's easier to see the loading screen
* Go to http://jetpack.cloud.localhost:3000/ and confirm that you can see a pulsating Jetpack logo while the Jetpack Cloud is being loaded

Asana card: 1156570014567299-as-1157678527369108

Commit:
Add Jetpack Cloud loading screen logo and animation

tyxla commented on client/document/index.jsx:
```diff
@@ -133,7 +137,11 @@ class Document extends React.Component {
 								<div className="layout__content">
-									<WordPressLogo size={ 72 } className="wpcom-site__logo" />
+									{ isJetpackCloudFlow ? (
+										<JetpackLogo size={ 72 } className="wpcom-site__logo" />
```
Should we use a different class name for the Jetpack Logo here? The `wpcom-site__logo` no longer makes sense with the Jetpack logo.

delawski replied:
It is intentional. I left the `wpcom-site__logo` class here since this element still [is contained within `wpcom-site` wrapper](https://github.com/Automattic/wp-calypso/pull/39172/files#diff-3fb371c23faa311830bf33b7e7e855a3R129) and logically belongs to it.

I wouldn't read it as a "WPCOM logo" but rather a "logo within WPCOM site (admin)". I know it is still quite confusing, but changing `wpcom-site` to something like `jetpack-cloud-site` just for Jetpack Cloud because of the logo would be an overkill in my opinion 🙂But I'm happy to hear what's your take on this.

Alternative solution is to break any relation between the `logo` and `wpcom-site` element for Jetpack Cloud and have a completely independent CSS element here like `jetpack-cloud-loader-logo`.

tyxla replied:
I'm happy with that rationale, @delawski. It's confusing without the explanation, but I'm okay to keep it as-is as a simple and straightforward solution.

tyxla commented on client/assets/stylesheets/jetpack-cloud.scss:
```diff
@@ -5,6 +5,19 @@
 // Global layout
 @import 'main';
+
+.is-section-jetpack-cloud .wpcom-site__logo {
```
Should we use a different class name for the Jetpack Logo here? The `wpcom-site__logo` no longer makes sense with the Jetpack logo.

And then also reuse all the styles that are defined on `.wpcom-site__logo` that we need.

delawski replied:
See my comment around the class name [in the other file](https://github.com/Automattic/wp-calypso/pull/39172/files#diff-3fb371c23faa311830bf33b7e7e855a3R141).

tyxla commented on client/document/index.jsx:
```diff
@@ -79,6 +80,9 @@ class Document extends React.Component {
 			'woocommerce-onboarding' === requestFrom;
 
+		const isJetpackCloudFlow =
+			config.isEnabled( 'jetpack-cloud' ) && 'jetpack-cloud' === sectionName;
```
Is the `config.isEnabled` check really useful here? I'd say the section name check is sufficient.

If you agree, maybe we could remove the `const` altogether and move the check inline as it's simple enough.

delawski replied:
It makes sense to me. I've based this check on the [existing check for `isJetpackWooCommerceFlow` a few lines above](https://github.com/Automattic/wp-calypso/pull/39172/files#diff-3fb371c23faa311830bf33b7e7e855a3R78), but - really - we can rely on the section name since it's determined by the same config flag anyway.

tyxla replied:
Not only we can rely on the section name, but it also will make sense if we ever decide to enable multiple sections for the Jetpack Cloud and we run it alongside Calypso sections.

tyxla commented on client/document/index.jsx:
```diff
@@ -79,6 +80,9 @@ class Document extends React.Component {
 			'jetpack-connect' === sectionName &&
 			'woocommerce-onboarding' === requestFrom;
 
+		const isJetpackCloudFlow =
```
Nitpicking: I'd suggest `section` vs `flow`, as Jetpack Cloud isn't really "flow".

```suggestion
		const isJetpackCloudSection
```

tyxla reviewed (changes requested):
This is working really well! I had to remove the `clientRender` from one of the routes to test it well as it loads too fast 😆 

Anyway, I've left some suggestions on the code, let me know what you think.

Commit:
Address review feedback

delawski commented:
Thank you @tyxla for the review. I've addressed your feedback in f4f7084.

In there, I've also addressed @eeeeevon13's design feedback about having a "Loading" message on the loading screen as well. Here's a preview of it:

![2020-01-31 12 36 56](https://user-images.githubusercontent.com/478735/73536816-26960280-4427-11ea-8cde-c9db25f96cf7.gif)

cc @keoshi

keoshi reviewed (approved):
Tested this in a multitude of screen widths and it looks great on all of them. 🚀

tyxla commented on client/document/index.jsx:
```diff
@@ -133,7 +136,14 @@ class Document extends React.Component {
 								<div className="masterbar" />
 								<div className="layout__content">
-									<WordPressLogo size={ 72 } className="wpcom-site__logo" />
+									{ 'jetpack-cloud' === sectionName ? (
```
Ideally, this component should be section-agnostic. This isn't the case right now even before introducing this change, so I'm fine accepting this. I just wanted to mention that.

tyxla reviewed (approved):
LGTM, thanks 🚢
